annotations/src/main/java/org/corfudb/runtime/object/ISMRObject.java

```diff
@@ -0,0 +1,13 @@
+package org.corfudb.runtime.object;
+
+
+/**
+ * All objects that need to be understood by Corfu have to implement this
+ * interface, whose purpose is to facilitate better integration between
+ * different Corfu components and the underlying object.
+ *
+ * @param <T> The underlying object type
+ */
+public interface ISMRObject<T>
```

Comment:
I think this needs to be reconciled with `ICorfuSMR`:  On master the `ISMRObject` is actually unnecessary and should have been collapesed with `ICorfuSMR` and concrete Classes should override hashcode, toString,etc.. if they need to.

Reply:
Yep, makes sense.

Reply:
Consolidated.